vltt/assignment2/Q1.cpp

```diff
@@ -0,0 +1,28 @@
+#include <iostream>
+#include <algorithm>
+#include <iomanip>
+#include <vector>
+
+struct node {
+    node *left = nullptr;
```

Comment:
[optional] If you have time, consider learning about [std::unique_ptr](http://www.drdobbs.com/cpp/c11-uniqueptr/240002708). This is the preferred way of using pointers in C++ and will be useful in future assignments, especially when you'll have to write tests and manage objects' memory :)